### PR TITLE
Return error on invalid YAML

### PR DESCRIPTION
**Pull request by JackuB**
This should fix false positive for invalid YAML files. 
It should also return error on invalid YAML.

### CODE REVIEW

**Comment by JackuB** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;PUO1sCWR&#x2F;65-yaml-errors-and-warnings-showing-in-the-editor"> YAML errors and warnings showing in the editor</a>

File: test/adapter.js [hunk 1]
```diff
@@ -104,6 +97,23 @@ describe('Swagger 2.0 adapter', () => {
     });
   });
 
+  context('cannot parse invalid Swagger YAML', () => {
+    const source = 'swagger: "2.0"\nbad: }';
+    let result;
+
+    before((done) => {
+      fury.parse({source}, (err) => {
+        result = err;
```
Comment (smizell) [inline test/adapter.js]: Is the correct naming of this variable? Wouldn't the result of the parse be called the `result`, but you are concerned with the `err` here?

**Comment by smizell** [comment]
@JackuB Just one small naming question.

File: test/adapter.js (hunk 1 above)
Comment (JackuB) [inline test/adapter.js]: I meant it as "a result of parse()", but let's give it a proper name.

**Comment by JackuB** [comment]
@smizell updated

**Pull request merged**